src/bundle/Controller/LocationController.php

```diff
@@ -335,6 +337,11 @@ public function trashAction(Request $request): Response
 
         if ($form->isSubmitted()) {
             $result = $this->submitHandler->handle($form, function (LocationTrashData $data) {
+                if (isset($data->getTrashOptions()[HasUniqueAssetRelation::TRASH_ASSETS])
+                    && HasUniqueAssetRelation::RADIO_SELECT_TRASH_WITH_ASSETS === $data->getTrashOptions()[HasUniqueAssetRelation::TRASH_ASSETS]
+                ) {
+                    $this->trashRelatedAsset($data->getLocation()->getContentInfo());
+                }
```

Comment:
1) This logic should be placed in `TrashLocationStrategy` or at least in  `\EzSystems\EzPlatformAdminUiBundle\Controller\LocationController::handleTrashLocationForm` method 
2) Location and assets related to content pointed by this location should be trashed in transaction